LAKMobile/src/screens/SignupScreen.tsx

```diff
@@ -76,13 +76,13 @@ export function SignupScreen({ navigation }: SignupProps) {
     auth.clearError();
     setSignupPressed(true);
 
-    if (
-      validateName() &&
-      validatePhone() &&
-      validateLocation() &&
-      validatePin() &&
-      validateConfirmPin()
-    ) {
+    validateName();
+    validatePhone();
+    validateLocation();
+    validatePin();
+    validateConfirmPin();
+
+    if (nameValid && phoneValid && locationValid && PINValid && confirmPINValid) {
```

Comment:
Good catch on the bug! I have a comment here:
Since setState doesn't update the state immediately after it's called  (there can sometimes be a little bit of lag), when we move to the if statement to check if the fields are valid, it might return false when we don't want it to. Instead, what we can do is read in the values returned by each of the validation functions and use those for our checking if flow since we can guarantee that the return values are the current return values